pkg/controller/controller.go

```diff
@@ -522,3 +523,39 @@ func isInstanceReady(instance *v1alpha1.Instance) bool {
 
 	return false
 }
+
+// IsOwnedBy checks if the given object has a controller ownerReference set to the given owner
+func IsOwnedBy(obj metav1.Object, owner metav1.Object) bool {
+	ref := GetControllerOf(obj)
+	if ref == nil {
+		return false
+	}
+	return ref.UID == owner.GetUID()
+}
+
+// GetControllerOf returns the controllerRef if controllee has a controller,
```

Comment:
I'd like to get @caesarxuchao's opinion on whether this code is correct, since he is the creator of ownerRefs.